### PR TITLE
treewide: support structuredAttrs in setup hooks (part 3)

### PR DESCRIPTION
**Pull request by wolfgangwalther**
## Description of changes

This is a second follow up to #318614, which had a lot more commits initially, similar to the first one in #335666. For context, please see those PRs.

For some of the commits in this PR, I was able to find some packages to test with/without structuredAttrs similar to the PRs before, but not all of them. The changes should be straight-forward by now anyway, using the other two PRs as reference.

After this PR, there are only a few more hooks left on my list, but I will open individual PRs for those later.

CC @emilazy @philiptaron @tie

Since the changed setup hooks won't trigger ofborg to mention the maintainers:

- swiftpm: @dduan @stephank @trepetti @trundle
- gradle: @lorenzleutgeb @liff
- xcbuild: @copumpkin @matthewbauer 
- cuda-modules: @NixOS/cuda-maintainers 
- extra-cmake-modules: ?
- *bsd: ? 

## Things done

- Built on platform(s)
  - [x] x86_64-linux (all except xcbuild, darwin.libutil)
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin (xcbuild, darwin.libutil)
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by reckenrode** [comment]
xcbuild changes look fine. They match (more or less) what I’ve done in my Darwin refactor branch.

File: pkgs/os-specific/bsd/setup-hook.sh [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# shellcheck shell=bash disable=SC2154,SC2164
```
Comment (philiptaron) [inline pkgs/os-specific/bsd/setup-hook.sh:1]: https://www.shellcheck.net/wiki/SC2154 -- where's this one trigger?

https://www.shellcheck.net/wiki/SC2164 -- where's this one trigger?

**Comment by philiptaron** [comment]
Adding @artemist for the *bsd bits. Please feel free to add whomever you think else might be the right person(s) to review!

**Review by philiptaron — APPROVED** [review]
The changes to all but *BSD look quite straightforward to me.

**Comment by artemist** [comment]
Nothing looks super out of place with BSD, though I haven't tested it. I think @rhelmot has dealt with the setup hook more

**Review by K900 — COMMENTED** [review]
ECM looks OK

File: pkgs/os-specific/bsd/setup-hook.sh (hunk 1 above)
Comment (philiptaron) [reply pkgs/os-specific/bsd/setup-hook.sh:1]: Thanks, I'll file that as assumed on future reviews.

**Comment by philiptaron** [comment]
Barring any objections, I'd like to merge this in a day or less.

**Pull request merged**